m3-sys/cminstall/src/config-no-install/AMD64_LINUX

```diff
@@ -1,10 +1,14 @@
 readonly TARGET = "AMD64_LINUX" % code generation target
 readonly GNU_PLATFORM = "amd64-linux" % "cpu-os" string for GNU
 
-SYSTEM_CC = "gcc -g -m64 -fPIC" % C compiler
-SYSTEM_CXXC = "g++ -g -m64 -fPIC" % C++ compiler
+SYSTEM_CC = "gcc -gdwarf-2 -gstrict-dwarf -m64 -fPIC" % C compiler
+SYSTEM_CXXC = "g++ -gdwarf-2 -gstrict-dwarf -m64 -fPIC" % C++ compiler
```

Comment:
```

J.K }} 
J.K }} 08.06.2021, 22:00, "Jay Krell" ***@***.***>:
J.K }} @jaykrell commented on this pull request.
J.K }} 
J.K }} 
J.K }} 
J.K }} In m3-sys/cminstall/src/config-no-install/AMD64_LINUX:
J.K }} > @@ -1,10 +1,14 @@
J.K }}  readonly TARGET = "AMD64_LINUX" % code generation target
J.K }}  readonly GNU_PLATFORM = "amd64-linux" % "cpu-os" string for GNU
J.K }}  
J.K }} -SYSTEM_CC = "gcc -g -m64 -fPIC" % C compiler
J.K }} -SYSTEM_CXXC = "g++ -g -m64 -fPIC" % C++ compiler
J.K }} +SYSTEM_CC = "gcc -gdwarf-2 -gstrict-dwarf -m64 -fPIC" % C compiler
J.K }} +SYSTEM_CXXC = "g++ -gdwarf-2 -gstrict-dwarf -m64 -fPIC" % C++ compiler
J.K }} 
J.K }} Are these changes needed?
J.K }} 
 
 
 About DWARF-2 I am not shue:



--- AMD64_LINUX 000	Thu May 20 12:11:48 2021
+++ AMD64_LINUX	Wed Jun 09 10:51:40 2021
@@ -6,5 +6,9 @@
 
 readonly SYSTEM_ASM = "as --64" % Assembler
 
+% PIE debugging requires gdb 7.1 or newer, even for C
+% see http://www.gnu.org/software/gdb/download/ANNOUNCEMENT
+readonly POSITION_INDEPENDENT_EXECUTABLE = "-no-pie"
+
 include("AMD64.common")
 include("Linux.common")



m3gdb_good_with_gcc_5.4_ -tui  -tui ./hello



{{{
...cal/cm3/cm3/examples/hello/AMD64_LINUX# m3gdb_good_with_gcc_5.4_ -tui  -tui ./hello
}}}




{{{
   ┌───────────────────────────────────────────────────────────────────────────────────┐
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                 [ No Source Available ]                                           │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   │                                                                                   │
   └───────────────────────────────────────────────────────────────────────────────────┘
exec No process In:                                                  Line: ??   PC: 0x0
welcome to change it and/or distribute copies of it under certain conditions.
Type "show copying" to see the conditions.
There is absolutely no warranty for GDB.  Type "show warranty" for details.
This GDB was configured as "x86_64-unknown-linux-gnu"...Using host libthread_db library
"/lib/x86_64-linux-gnu/libthread_db.so.1".

(m3gdb) break main
Breakpoint 1 at 0x4009f6
(m3gdb) run1 at 0x4009f6
}}}




{{{
                                               BFD: /usr/lib/x86_64-linux-gnu/libc.so.6:
 invalid relocation type 37
                           BFD: /usr/lib/x86_64-linux-gnu/libc.so.6: invalid relocation
type 37
       BFD: /usr/lib/x86_64-linux-gnu/libc.so.6: invalid relocation type 37
                                                                           BFD: /usr/lib
/x86_64-linux-gnu/libc.so.6: invalid relocation type 37
                                                       BFD: /usr/lib/x86_64-linux-gnu/li
bc.so.6: invalid relocation type 37
                     [ No Source Available ]/lib/x86_64-linux-gnu/libc.so.6: invalid rel
ocation type 37
               BFD: /usr/lib/x86_64-linux-gnu/libc.so.6: invalid relocation type 37
                                                                                   BFD:
/usr/lib/x86_64-linux-gnu/libc.so.6: invalid relocation type 37
                                                               BFD: /usr/lib/x86_64-linu
x-gnu/libc.so.6: invalid relocation type 37
                                           BFD: /usr/lib/x86_64-linux-gnu/libc.so.6: inv
alid relocation type 37
                       BFD: /usr/lib/x86_64-linux-gnu/libc.so.6: invalid relocation type
multi-thre Thread 1404271 In: main                              Line: ??   PC: 0x4009f6
o.6: invalid relocation type 37
                               BFD: /usr/lib/x86_64-linux-gnu/libc.so.6: invalid relocat
ion type 37
           BFD: /usr/lib/x86_64-linux-gnu/libc.so.6: invalid relocation type 37
[New Thread 140427196317952 (LWP 32440)]
[Switching to Thread 140427196317952 (LWP 32440)]
[Switching to Thread 140427196317952 (LWP 32440)]
Breakpoint 1, 0x00000000004009f6 in main ()2440)]
(m3gdb) nt 1, 0x00000000004009f6 in main ()2440)]
}}}




{{{
      ../src/runtime/common/RTLinker.m3        BFD: /usr/lib/x86_64-linux-gnu/libc.so.6:
   │25                                                                                 │
   │26      PROCEDURE InitRuntime (p_argc: INTEGER;  p_argv, p_envp, p_instance: ADDRES│
   │27        (*Note: This procedure is called BEFORE any modules are linked! *)       │
   │28        BEGIN                                                                    │
  >│29          IF init_done THEN RETURN; END;                                         │
   │30          init_done := TRUE;                                                     │
   │31                                                                                 │
   │32          (* make sure we can at least reference our own interface variables! *) │
   │33          FixImports (RTLinkerX.RTLinker_M3 (0));                                │
   │34                                                                                 │
   │35          (* expose the global environment *)                                    │
   │36          argc := p_argc;                                                        │
   │37          argv := p_argv;                                                        │
   │38          envp := GetEnvironmentStrings (p_envp);                                │
   │39          instance := p_instance;                                                │
   │40                                                                                 │
   │41          (* PrintModule(RTLinkerX.RTHooks_M3(0)); *)                            │
                       BFD: /usr/lib/x86_64-linux-gnu/libc.so.6: invalid relocation type
multi-thre Thread 1404271 In: InitRuntime                 Line: 29   PC: 0x7fb7c165c07d
[Switching to Thread 140427196317952 (LWP 32440)]
Breakpoint 1, 0x00000000004009f6 in main ()2440)]
(m3gdb) step, 0x00000000004009f6 in main ()2440)]
Single stepping until exit from function main, )]
which has no line number information.ion main, )]
InitRuntime (p_argc=1, p_argv=16_00007ffdb68f82f8, p_envp=16_00007ffdb68f8308,
    p_instance=NIL) at ../src/runtime/common/RTLinker.m3:296_00007ffdb68f8308,
Current language:  auto; currently Modula-3n/RTLinker.m3:296_00007ffdb68f8308,
(m3gdb) stance=NIL) at ../src/runtime/common/RTLinker.m3:296_00007ffdb68f8308,
}}}



```

Reply:
```

J.K }} 
J.K }} 08.06.2021, 22:00, "Jay Krell" ***@***.***>:
J.K }} @jaykrell commented on this pull request.
J.K }} 
J.K }} 
J.K }} 
J.K }} In m3-sys/cminstall/src/config-no-install/AMD64_LINUX:
J.K }} > @@ -1,10 +1,14 @@
J.K }}  readonly TARGET = "AMD64_LINUX" % code generation target
J.K }}  readonly GNU_PLATFORM = "amd64-linux" % "cpu-os" string for GNU
J.K }}  
J.K }} -SYSTEM_CC = "gcc -g -m64 -fPIC" % C compiler
J.K }} -SYSTEM_CXXC = "g++ -g -m64 -fPIC" % C++ compiler
J.K }} +SYSTEM_CC = "gcc -gdwarf-2 -gstrict-dwarf -m64 -fPIC" % C compiler
J.K }} +SYSTEM_CXXC = "g++ -gdwarf-2 -gstrict-dwarf -m64 -fPIC" % C++ compiler
J.K }} 
J.K }} Are these changes needed?
J.K }} 
 

 With "full patch":

==
-SYSTEM_CC = "gcc -g -m64 -fPIC" % C compiler
-SYSTEM_CXXC = "g++ -g -m64 -fPIC" % C++ compiler
+SYSTEM_CC = "gcc -gdwarf-2 -gstrict-dwarf -m64 -fPIC" % C compiler
+SYSTEM_CXXC = "g++ -gdwarf-2 -gstrict-dwarf -m64 -fPIC" % C++ compiler
==



result is "more usefull". See:


{{{
      ../src/runtime/common/RTLinker.m3─       BFD: /usr/lib/x86_64-linux-gnu/libc.so.6:
   │77        VAR imp: RT0.ImportPtr;                                                  │
   │78        BEGIN                                                                    │
   │79          IF (m = NIL) THEN RETURN; END;                                         │
   │80          TraceModule("FixImports: ", m);                                        │
   │81          imp := m.imports;                                                      │
   │82          WHILE (imp # NIL) DO                                                   │
   │83            IF (imp.import = NIL) THEN  imp.import := imp.binder (0);  END;      │
  >│84            imp := imp.next;                                                     │
   │85          END;                                                                   │
   │86        END FixImports;                                                          │
   │87                                                                                 │
   │88      (*-------------------------------------------------------- module linking -│
   │89                                                                                 │
   │90      CONST                                                                      │
   │91        LS_Initial = 0;                                                          │
   │92        LS_Linked  = 1;                                                          │
   │93        LS_TypesOK = 2;                                                          │
                       BFD: /usr/lib/x86_64-linux-gnu/libc.so.6: invalid relocation type
multi-thre Thread 1396864 In: FixImports                        Line: 84   PC: 0x75bfb7
1: imp = 16_0000000000923d28
1: imp = 16_0000000000923d28
RTLinkerX (mode=0) at ../src/runtime/common/RTLinkerX.i3:80
1: imp = warning: This could be the wrong instance of procedure FixImports, if a proc pa
ram with a nested procedure value was called.TLinkerX.i3:80
16_0000000000923d28rocedure value was called.TLinkerX.i3:80
FixImports (m=16_0000000000923b60) at ../src/runtime/common/RTLinker.m3:84
1: imp = 16_0000000000923d2823b60) at ../src/runtime/common/RTLinker.m3:84
(m3gdb)  16_0000000000923d2823b60) at ../src/runtime/common/RTLinker.m3:84
}}}




```